src/parseChangelog.test.js

```diff
@@ -0,0 +1,501 @@
+    expect(changelog.getReleaseChanges('0.0.2')).toStrictEqual({
+      Fixed: ['Something'],
+    });
+    expect(changelog.getReleaseChanges('0.0.1')).toStrictEqual({
+      Changed: ['Something'],
+    });
+    expect(changelog.getUnreleasedChanges()).toStrictEqual({});
+  });
+
+  it('should parse changelog with release statuses', () => {
+    const changelog = parseChangelog({
+      changelogContent: outdent`
+        # Changelog
+        All notable changes to this project will be documented in this file.
+
+        The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+        and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+        ## [Unreleased]
+
+        ## [1.0.0] - 2020-01-01 [BETA]
+        ### Changed
+        - Something else
+
+        ## [0.0.2] - 2020-01-01 [WITHDRAWN]
+        ### Fixed
+        - Something
+
+        ## [0.0.1] - 2020-01-01 [DEPRECATED]
+        ### Changed
+        - Something
+
+        [Unreleased]: https://github.com/ExampleUsernameOrOrganization/ExampleRepository/compare/v1.0.0...HEAD
+        [1.0.0]: https://github.com/ExampleUsernameOrOrganization/ExampleRepository/compare/v0.0.2...v1.0.0
+        [0.0.2]: https://github.com/ExampleUsernameOrOrganization/ExampleRepository/compare/v0.0.1...v0.0.2
+        [0.0.1]: https://github.com/ExampleUsernameOrOrganization/ExampleRepository/releases/tag/v0.0.1
+        `,
+      repoUrl:
+        'https://github.com/ExampleUsernameOrOrganization/ExampleRepository',
+    });
+
+    expect(changelog.getReleases()).toStrictEqual([
+      { date: '2020-01-01', status: 'BETA', version: '1.0.0' },
+      { date: '2020-01-01', status: 'WITHDRAWN', version: '0.0.2' },
+      { date: '2020-01-01', status: 'DEPRECATED', version: '0.0.1' },
+    ]);
+  });
+
+  it('should throw if Unreleased header is missing', () => {
+    const brokenChangelog = outdent`
+      # Changelog
+      All notable changes to this project will be documented in this file.
+
+      The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+      and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+      [Unreleased]: https://github.com/ExampleUsernameOrOrganization/ExampleRepository/
+      `;
+    expect(() =>
+      parseChangelog({
+        changelogContent: brokenChangelog,
+        repoUrl:
+          'https://github.com/ExampleUsernameOrOrganization/ExampleRepository',
+      }),
+    ).toThrow('Failed to find Unreleased header');
+  });
+
+  it('should throw if Unreleased reference link definition is missing', () => {
+    const brokenChangelog = outdent`
+      # Changelog
+      All notable changes to this project will be documented in this file.
+
+      The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+      and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+      ## [Unreleased]
```

Comment:
some of our changelogs dont have the link definition:

https://github.com/MetaMask/inpage-provider/blob/main/CHANGELOG.md#unreleased

should we just change those to add it?